### PR TITLE
[Bundle products in order] Update design of child items in product bundle on order creation screen

### PR DESCRIPTION
**Pull request by rachelmcr**
Part of: #11250
Part of: #10428

## Description
We want to make some design updates to the product cards on the order creation screen. These are mainly focused on improving the design of child items for product bundles, but some of the changes apply to all product cards, to clean up the overall design:

* For all cards, when the card is expanded:
   * Reduce the product details font size to subheadline with black font in the top section
   * Keep the row with stock details in the top section
   * Replace the price with the SKU in the top section (the price is only shown below, in the expanded details)
   * Remove the top divider (from https://github.com/woocommerce/woocommerce-ios/pull/11268#pullrequestreview-1747159094)
* For child items:
   * Indented 40×40pt thumbnails
   * Reduce the product name font size to subheadline
   * Change the chevron colour to grey

In a separate PR, the child items will be updated to be read-only (to remove options to add discount, remove from order).

## Testing instructions
Prerequisite: the store has [Product Bundles extension](https://woocommerce.com/products/product-bundles/) enabled, and also a bundle product with non-empty items.

- Go to the Orders tab
- Tap `+` to create an order
- Tap `Add Products` and tap a non-bundle product
- Tap a bundle product from the prerequisite
- Configure the bundle
- Add the products to the order
- On the order form, confirm the design is updated with the changes listed above (with nothing looking out of place for bundle or non-bundle products)

## Screenshots
/|Before|After
-|-|-
Collapsed|Before-Collapsed|Design-After-Collapsed
Expanded - Non-bundle product|Before-Product-Expanded|Design-After-Product-Expanded
Expanded - Bundle product|Before-ProductBundle-Expanded|Design-After-ProductBundle-Expanded

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/CollapsibleView.swift [hunk 1]
```diff
@@ -46,6 +49,7 @@ struct CollapsibleView<Label: View, Content: View>: View {
                     Spacer()
                     if isCollapsible {
                         Image(uiImage: isCollapsed ? .chevronDownImage : .chevronUpImage)
+                            .foregroundColor(hasSubtleChevron ? Color(.textSubtle) : .black)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/CollapsibleView.swift]: From the design w3JBReW1UJesCf02sj43kR-fi-982_5822, it looks like the chevron is `accent` color for the parent view? If `black` is the intended color, using `text` color would look better in dark mode.

Screenshot 2023-11-28 at 9 29 53 AM

```suggestion
                            .foregroundColor(hasSubtleChevron ? Color(.textSubtle) : .accent)
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductCard.swift [hunk 2]
```diff
@@ -138,31 +134,40 @@ private struct CollapsibleProductRowCard: View {
...
-                                          productImageSize: Layout.productImageSize,
+                                          productImageSize: viewModel.hasParentProduct ? Layout.childProductImageSize : Layout.parentProductImageSize,
                                           scale: scale,
                                           productImageCornerRadius: Layout.productImageCornerRadius,
                                           foregroundColor: Color(UIColor.listSmallIcon))
+                    .padding(.leading, viewModel.hasParentProduct ? Layout.childLeadingPadding : 0)
                     VStack(alignment: .leading) {
                         Text(viewModel.name)
+                            .font(viewModel.hasParentProduct ? .subheadline : .none)
+                            .foregroundColor(.black)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductCard.swift]: `text` color might work better in dark mode
```suggestion
                            .foregroundColor(.text)
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductCard.swift [hunk 3]
```diff
@@ -138,31 +134,40 @@ private struct CollapsibleProductRowCard: View {
...
                         Text(viewModel.stockQuantityLabel)
-                            .foregroundColor(.secondary)
-                            .renderedIf(isCollapsed)
+                            .font(.subheadline)
+                            .foregroundColor(isCollapsed ? Color(.textSubtle) : .black)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductCard.swift]: similar to above, `text` might work better in dark mode
```suggestion
                            .foregroundColor(isCollapsed ? Color(.textSubtle) : .text)
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductCard.swift [hunk 4]
```diff
@@ -138,31 +134,40 @@ private struct CollapsibleProductRowCard: View {
...
                         Text(viewModel.skuLabel)
-                            .foregroundColor(.secondary)
+                            .font(.subheadline)
+                            .foregroundColor(.black)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductCard.swift]: similar to above, `text` might work better in dark mode
```suggestion
                            .foregroundColor(.text)
```

**Review by jaclync — APPROVED** [review]
Looks nice! 🚢  just some suggestions for dark mode.

I understand we don't have a design for dark mode, feel free to deviate from the suggestions to make it look better.

collapsed | expanded
-- | --
 |

File: WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/CollapsibleView.swift (hunk 1 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/CollapsibleView.swift]: Good catch! I missed that change to `accent` here.

**Comment by rachelmcr** [comment]
Thanks for catching those dark mode issues! 🤦‍♀️  Using `text` works great here.

**Pull request merged**